### PR TITLE
HB-6620: Per-partner consent API

### PR DESCRIPTION
**Pull request by daniel-barros**
Add support for upcoming partnerConsentStatus API.

WARNING: DO NOT MERGE UNTIL THE CORRESPONDING CORE SDK VERSION HAS BEEN RELEASED.

### CODE REVIEW

**Review by CB-HaoxinLi — COMMENTED** [review]
This adapter has some business logic for calling `onPartnerConsentStatusChange()`. Can we add some unit tests against them?

`UsercentricsAdapter` is more than a simply SDK wrapper and it deserves unit tests for other business logic as well. I thinking about wrapping Usercentrics with a custom protocol so that we can create a Usercentrics mock for unit test, which will take significant efforts that beyond this PR. I think this PR looks good, and the unit tests can be added later.

File: Source/UsercentricsAdapter.swift [hunk 1]
```diff
@@ -14,7 +14,7 @@ import UsercentricsUI
 /// Chartboost Core Consent Usercentrics adapter.
 @objc(CBCUsercentricsAdapter)
 @objcMembers
-public final class UsercentricsAdapter: NSObject, ConsentAdapter {
+public final class UsercentricsAdapter: NSObject, InitializableModule, ConsentAdapter {
```
[the remark this answers is not in the document]
Comment (CB-HaoxinLi) [reply Source/UsercentricsAdapter.swift:17]: Ah, just saw the commit message saying this is for Objective C.

**Review by CB-HaoxinLi — APPROVED** [review]

**Comment by daniel-barros** [comment]
> This adapter has some business logic for calling `onPartnerConsentStatusChange()`. Can we add some unit tests against them?
> 
> `UsercentricsAdapter` is more than a simply SDK wrapper and it deserves unit tests for other business logic as well. I thinking about wrapping Usercentrics with a custom protocol so that we can create a Usercentrics mock for unit test, which will take significant efforts that beyond this PR. I think this PR looks good, and the unit tests can be added later.

My experience working on tests for other adapters years ago tells me it's not going to be worth it, but I'm not against it if you'd like us to add some unit tests. Once thing I'd ask is that we don't wrap Usercentrics in a protocols, as that will hide deprecation warnings for the methods we use.

**Review by CB-HaoxinLi — APPROVED** [review]
LGTM. Thanks for the notes about deprecation warnings, definitely need to watch out for them.

**Comment by CB-HaoxinLi** [comment]
I think we should change the base branch from `main` to `prerelease` now with 0.3.0 released.

**Comment by daniel-barros** [comment]
> I think we should change the base branch from `main` to `prerelease` now with 0.3.0 released.

This branch needs to be merged to main.
I created and pushed the prerelease branch just now off of this one

**Review by kendallrogers — APPROVED** [review]

**Pull request merged**